server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java

```diff
@@ -359,13 +371,54 @@ private SearchWsResponse doHandle(SearchRequest request, Request wsRequest) {
     // Must be done after loading of data as the "hidden" facet "debt"
     // can be used to get total debt.
     facets = reorderFacets(facets, options.getFacets());
+    replaceRuleIdsByRuleKeys(facets, data.getRules() == null ? emptyList() : data.getRules());
 
     // FIXME allow long in Paging
     Paging paging = forPageIndex(options.getPage()).withPageSize(options.getLimit()).andTotal((int) result.getHits().getTotalHits());
 
     return searchResponseFormat.formatSearch(additionalFields, data, paging, facets);
   }
 
+  private void replaceRuleIdsByRuleKeys(@Nullable Facets facets, List<RuleDefinitionDto> alreadyLoadedRules) {
+    if (facets == null || facets.get(PARAM_RULES) == null) {
+      return;
+    }
+
+    // The facet for PARAM_RULES contains the id of the rule as the key
+    // We need to update the key to be a RuleKey
+    LinkedHashMap<String, Long> rulesFacet = facets.get(PARAM_RULES);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      Set<String> ruleKeysToLoad = new HashSet<>();
+      ruleKeysToLoad.addAll(rulesFacet.keySet());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "rulesFacet" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Feh%2FSONAR-10303&issues=AWFBQzvZmckxG3vItnLG&open=AWFBQzvZmckxG3vItnLG)

[ISSUE_KEY]: AWFBQzvZmckxG3vItnLG

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED